src/kubectl_application_shell/app.py

```diff
@@ -72,46 +148,161 @@ def main(
-                        "command": [command],
-                        "args": args if args else [],
-                        "env": env,
-                        "envFrom": env_from,
-                        "resources": resources,
-                        "stdin": True,
-                        "stdinOnce": True,
-                        "tty": True,
-                        "volumeMounts": volume_mounts,
-                    }
-                ],
-                "volumes": volumes,
-            },
+    volumes = deployment_info["spec"]["template"]["spec"].get("volumes", [])
+    
+    # Add PVC mount if specified
+    if mount_pvc:
+        pvc_volume = {
+            "name": "debug-pvc",
+            "persistentVolumeClaim": {
+                "claimName": mount_pvc
+            }
         }
-    )
+        pvc_mount = {
+            "name": "debug-pvc",
+            "mountPath": "/mnt/data"
+        }
+        volumes.append(pvc_volume)
+        volume_mounts.append(pvc_mount)
+    
+    # Convert back to None if empty for clean JSON
+    volume_mounts = volume_mounts if volume_mounts else None
+    volumes = volumes if volumes else None
+    # Build container spec
+    container_spec = {
+        "name": container_name,
+        "image": image,
+        "command": [command],
+        "args": args if args else [],
+        "env": env,
+        "envFrom": env_from,
+        "resources": resources,
+        "stdin": True,
+        "stdinOnce": True,
+        "tty": not no_tty,
+        "volumeMounts": volume_mounts,
+    }
+    
+    # Add working directory if specified
+    if working_directory:
+        container_spec["workingDir"] = working_directory
+    
+    # Add security context if specified
+    security_context = {}
+    if run_as_user is not None:
+        security_context["runAsUser"] = run_as_user
+    if privileged:
+        security_context["privileged"] = True
+    if security_context:
+        container_spec["securityContext"] = security_context
+    
+    # Build pod spec
+    pod_spec = {
+        "spec": {
+            "containers": [container_spec],
+            "volumes": volumes,
+        },
+    }
+    
+    # Add service account if specified
+    if service_account:
+        pod_spec["spec"]["serviceAccountName"] = service_account
+    
+    # Add node selector if specified
+    if node_selector and "=" in node_selector:
+        key, value = node_selector.split("=", 1)
+        pod_spec["spec"]["nodeSelector"] = {key: value}
+    
+    # Add host network if specified
+    if host_network:
+        pod_spec["spec"]["hostNetwork"] = True
+    
+    # Add session tracking labels
+    labels = {
+        "app": "kubectl-application-shell",
+        "debug-user": get_user_identifier(),
+        "debug-deployment": deployment,
+    }
+    
+    # Add NetworkPolicy targeting label if needed
+    if network_policy:
+        labels["debug-pod"] = pod_name
+    
+    pod_spec["metadata"] = {"labels": labels}
```

Comment:
Also this is used ahead of being defined in https://github.com/tablecheck/kubectl-application-shell/pull/24/files#diff-065b3beb658a1a078c181cd4cabdfc8218ff1a3426c15d736f4121e449c706d6R270